jgrapht-core/src/main/java/org/jgrapht/alg/RankingPathElement.java

```diff
@@ -78,6 +83,35 @@ public double getWeight()
     {
         return (RankingPathElement<V, E>) super.getPrevPathElement();
     }
+
+    @Override
+    public Graph<V, E> getGraph()
+    {
+        return this.graph;
+    }
+
+    @Override
+    public V getStartVertex()
+    {
+        if (getPrevPathElement() == null) {
+            return super.getVertex();
+        }
+        return getPrevPathElement().getStartVertex();
```

Comment:
This could result in `StackOverflowException` if the path is long enough. 
Better to write this using a while loop.